sdk/core/azure-core-test/src/main/java/com/azure/core/test/InterceptorManager.java

```diff
@@ -62,6 +70,12 @@ public class InterceptorManager implements AutoCloseable {
     // Stores a map of all the HTTP properties in a session
     // A state machine ensuring a test is always reset before another one is setup
     private final RecordedData recordedData;
+    private final boolean enableTestProxy;
+    private TestProxyRecordPolicy testProxyRecordPolicy;
+    private TestProxyPlaybackClient testProxyPlaybackClient;
+    private final Queue<String> proxyVariableQueue = new LinkedList<>();
+    private List<TestProxySanitizer> recordSanitizers;
+    private List<TestProxyMatcher> customMatcher;
```

Comment:
Will be naive on this, what's the difference between a sanitizer and a matcher?

Reply:
Sanitizers are basically to redact sensitive information when recording. 
And Matchers are configurations/customizations that need to be matched when playing back the recording.

For example, Auth header will be REDACTED (Sanitization) whereas lets say when playing back we want to skip matching the `createdTime` we specify so in the matcher config.

Reply:
This is the language the test proxy uses, which led to us just bringing it in so we're consistent.